packages/api/src/controllers/asset.ts

```diff
@@ -288,15 +310,15 @@ app.use(
     };
 
     if (Array.isArray(data)) {
-      return data.map(toExternalAsset);
+      return Promise.all(data.map(toExternalAsset));
     }
     if ("id" in data) {
       return toExternalAsset(data);
```

Comment:
Shouldn't also this be `await toExternalAsset(data)` ?

Reply:
On JS you can wrap promises like `Promise<Promise<Promise<string>>>` and they will still work the same. 
i.e. you can also return a promise from an async function, and that will still work for the caller. 

note1 : TypeScript would complain if this wasn't valid, so we have an extra guarantee there 😌 
note2: Maybe the "promise unwrap" only happens for `async` functions. Maybe if you actually have a variable that is a `Promise<Promise<string>>` you'd have to manually unwrap it into a `Promise<string>` somehow. IDK. Maybe it's just magic tho lol